go/vt/topo/validator.go

```diff
@@ -0,0 +1,30 @@
+package topo
+
+import (
+	vtrpcpb "vitess.io/vitess/go/vt/proto/vtrpc"
+	"vitess.io/vitess/go/vt/vterrors"
+)
+
+// ValidateObjectName checks that the name is a valid object name.
+// Object names are used for things like keyspace and shard names
+// and must match specific constraints.
+// They are only allowed to use ASCII letters or digits, - and _.
+// No spaces or special characters are allowed.
+func validateObjectName(name string) error {
```

Comment:
@derekperkins @arthurschreiber @timvaillancourt @hkdsun I'd love to also explicitly solicit here feedback from outside PlanetScale to see if this suggest validation is problematic for anyone in the community and if we should further loosen it with more known to work characters or character classes or not then.

Reply:
@dbussink agreed on the "allowlist" approach

> // They are only allowed to use ASCII letters or digits, - and _.
> // No spaces or special characters are allowed.

Sounds like a reasonable limitation to me 👍

Reply:
Seems reasonable. I'd check with some of the Chinese adopters, as they're more likely to have utf8 characters.

Reply:
@derekperkins Do you know their GitHub handles? We can ping them here as well I think :smile:.

Reply:
We have a similar validation on keyspace names on our side. We only use Kubernetes and DNS friendly names 👍

Reply:
Sounds fine to me as well! 👍